src/main/java/usersandpostsCRUD/demo/service/PostService.java

```diff
@@ -0,0 +1,50 @@
+package usersandpostsCRUD.demo.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import usersandpostsCRUD.demo.dto.UserDto;
+import usersandpostsCRUD.demo.entity.Post;
+import usersandpostsCRUD.demo.entity.User;
+import usersandpostsCRUD.demo.exception.PostNotFoundException;
+import usersandpostsCRUD.demo.repository.PostRepository;
+import usersandpostsCRUD.demo.repository.UserRepository;
+
+import java.util.List;
+
+@Service
+public class PostService {
+
+    private final PostRepository postRepository;
+    private final UserRepository userRepository;
+
+    @Autowired
+    public PostService(PostRepository postRepository, UserRepository userRepository) {
+        this.postRepository = postRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Post> getAllPosts() {
+        return postRepository.findAll();
+    }
+
+    public Post createPost(Post post, long userId) {
+        userRepository.findById(userId).ifPresent(post::setUser);
```

Comment:
What happens if the user you look for in the dabase does not exist?
You cant create a post, but you can throw an exception for the user not existing.